src/cpp/lpnamer/model/ActiveLinks.cpp

```diff
@@ -11,22 +12,33 @@ bool doubles_are_different(const double a, const double b) {
 void ActiveLinksBuilder::addCandidate(const CandidateData& candidate_data) {
   unsigned int indexLink = getLinkIndexOf(candidate_data.link_id);
   _links.at(indexLink).addCandidate(
-      candidate_data, getProfileFromProfileMap(candidate_data.link_profile));
+      candidate_data, getProfilesFromProfileMap(candidate_data.direct_link_profile));
 }
 
 ActiveLinksBuilder::ActiveLinksBuilder(
-    const std::vector<CandidateData>& candidateList,
-    const std::map<std::string, LinkProfile>& profile_map)
-    : _candidateDatas(candidateList), _profile_map(profile_map) {
+    std::vector<CandidateData>  candidateList,
+    std::map<std::string, std::vector<LinkProfile>>  profile_map,
```

Comment:
I understand that you're using `std::move` to grab ownership to the object and avoid copies. However, the copy constructor is called when passing arguments by value. I suggest passing value by non-const reference instead, so that only the move constructor is called.

Pass args by value :
1. The copy constructor is called
2. The move constructor is called

Pass args by non-const reference
1. The move constructor is called

```cpp
#include <iostream>

struct Example {
  Example() = default;
  Example(Example& ex) {
    std::cout << "Example(Example& ex) COPY NOT GOOD\n";
  }
  Example(const Example& ex) {
    std::cout << "Example(const Example& ex)\n";
  }
  Example(Example&& ex) {
    std::cout << "Example(Example&& ex)\n";
  }
};

struct ExampleGrabByValue {
  ExampleGrabByValue(Example ex) : ex(std::move(ex)) {}
  Example ex;
};

struct ExampleGrabByRef {
  ExampleGrabByRef(Example& ex) : ex(std::move(ex)) {}
  Example ex;
};

int main() {
  Example ex;
  std::cout << "ExampleGrabByValue\n";
  ExampleGrabByValue byValue(ex);
  std::cout << "ExampleGrabByRef\n";
  ExampleGrabByRef byRef(ex);

  // Possible output
  // ExampleGrabByValue
  // Example(Example& ex) COPY NOT GOOD
  // Example(Example&& ex)
  // ExampleGrabByRef
  // Example(Example&& ex)
}
```

Reply:
It is "intented".
With the existing design the different values _candidateList_ or _profile_map_ are copied in class members. We need this copy (or need to make more profound changes).
The change was mainly prompted by sonar lint but it add expressiveness to the interface.

`void foo(const Bar& bar)` means that I take a read only reference to a existing value and hide the fact that we will copy it for ourself.
`void foo(Bar bar)`means that I need a value and express that if it already exist will at least be copied on the stack.